packages/bpk-component-modal/src/BpkModalV2/BpkModal.tsx

```diff
@@ -116,15 +116,27 @@ export const BpkModalV2 = (props: Props) => {
       ref.current?.showModal?.();
 
       const dialog = document.getElementById(`${id}`);
-      if (dialog) {
-        dialog.addEventListener('click', (event: Event) => {
-          const { target } = event;
-
-          if (target instanceof HTMLElement && target.id === `${id}`) {
-            ref.current?.close?.();
-          }
-        });
-      }
+      const dialogWithPolyfill = document.getElementById(`${id}-polyfill`);
+
+      const handleBackdropClick = (modal: HTMLElement | null) => {
+        if (modal) {
+          modal.addEventListener('click', (event: Event) => {
+            const { target } = event;
+
+            if (target === modal) {
+              if (modal === dialog) {
+                ref.current?.close?.();
+              }
+              if (modal === dialogWithPolyfill) {
+                onClose();
+              }
+            }
+          });
+        }
+      };
+
+      handleBackdropClick(dialog);
+      handleBackdropClick(dialogWithPolyfill);
```

Comment:
Just realised I had an issue when clicking the backdrop for the polyfill. It was not working. So did a wee bit of changes to handle the click when dialog is supported or not.